src/Command.ts

```diff
@@ -7,16 +7,31 @@ import {makeArray} from './util';
 /** Check if requirements are met. */
 // TODO this interface is ugly
 async function fulfillsRequirements (requirements: CommandRequirements, msg: Eris.Message, args: string[], ctx: CommandContext) {
-	const {owner, permissions, custom} = requirements;
+	const {owner, guildOnly, dmOnly, permissions, custom} = requirements;
 	const {client} = ctx;
+
 	// Owner checking
 	if (owner && client.app && client.app.owner.id !== msg.author.id) {
 		return false;
 	}
+
+	// Guild-only commands
+	if (guildOnly) {
+		if (!(msg.channel instanceof Eris.GuildChannel)) {
+			return false;
+		}
+	}
+
+	// DM-only commands
+	if (dmOnly) {
+		if (!(msg.channel instanceof Eris.PrivateChannel)) {
+			return false;
+		}
+	}
+
 	// Permissions
 	if (permissions && permissions.length > 0) {
-		// If we require permissions, the command can't be used in direct
-		// messages
+		// Permission checks only make sense in guild channels
 		if (!(msg.channel instanceof Eris.GuildChannel)) {
```

Comment:
```suggestion
		if (!msg.guildID) {
```

Reply:
This one actually does need to be kept as an `instanceof` check because we rely on `channel.permissionsOf` below, which doesn't exist on private channels. Will be reverted in the next commit.